clients/gateway/gateway.go

```diff
@@ -0,0 +1,101 @@
+package gateway
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/NethermindEth/juno/utils"
+)
+
+type Client struct {
+	url    string
+	client *http.Client
+	log    utils.SimpleLogger
+}
+
+func NewClient(url string, log utils.SimpleLogger) *Client {
+	url = strings.TrimRight(url, "/")
+	return &Client{
+		url:    url,
+		client: http.DefaultClient,
+		log:    log,
+	}
+}
+
+type BroadcastInvokeTxn struct {
+	Type          string   `json:"type"`
+	Version       string   `json:"version"`
+	MaxFee        string   `json:"max_fee"`
+	Signature     []string `json:"signature"`
+	Nonce         string   `json:"nonce"`
+	SenderAddress string   `json:"sender_address"`
+	CallData      []string `json:"calldata"`
+}
```

Comment:
I think these all should be field elements according to the spec. If we are going to unmarshal the Txn into a type rather than just treating as a JSON blob, let's use the actual types of the fields.